lib/GILGen/Context.hpp

```diff
@@ -110,6 +110,16 @@ class Context {
         return bb;
     }
 
+    // Overload allowing block arguments (for SSA-like value merging)
+    gil::BasicBlock *buildBB(
+        llvm::StringRef name, llvm::ArrayRef<gil::Type> argTypes
+    )
```

Comment:
you can merge this function with the previous one by using an optional second argument